File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    citations.sort(reverse = True)
    citation_max = citations[0]
    
    while citation_max >= 0:
        count = 0
        for citation in citations:
            if citation >= citation_max:
                count += 1
            if citation < citation_max:
                break
        
        if count >= citation_max:
            return citation_max
        
        citation_max -= 1

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
        int citationMax = citations[citations.length-1];
        
        while (citationMax > 0) {
            int count = 0;
            for (int c: citations) {
                if (c >= citationMax) {
                    count += 1;
                }
            }
            
            if (count >= citationMax) {
                break;
            }

            citationMax -= 1;
        }
        
        return citationMax;
    }
}